src/authentificationHelper.js

```diff
@@ -11,38 +17,156 @@ export const getCredentials = async (serverURL) => {
   }
 }
 
+/**
+ * Store the credentials in the storage
+ * @param {String} serverURL Server URL to save the credentials for. Will be used as key.
+ * @param {String} username Username to save
+ * @param {String} password Password to save /!\ Password will be stored in plain text /!\
+ */
 export const storeCredentials = async (serverURL, username, password) => {
   const serverCredentialDictAsJSON = await AsyncStorage.getItem('credentials')
   let serverCredentialDict = JSON.parse(serverCredentialDictAsJSON)
   serverCredentialDict = { ...serverCredentialDict, [serverURL]: { username, password } }
   await AsyncStorage.setItem('credentials', JSON.stringify(serverCredentialDict))
 }
 
+/**
+ * Remove the credentials from the storage
+ * @param {String} serverURL Server URL to delete the credentials for.
+ */
 export const removeCredentials = async (serverURL) => {
   const serverCredentialDictAsJSON = await AsyncStorage.getItem('credentials')
   const serverCredentialDict = JSON.parse(serverCredentialDictAsJSON)
   delete serverCredentialDict[serverURL]
   await AsyncStorage.setItem('credentials', JSON.stringify(serverCredentialDict))
 }
 
+/**
+ * Try to connect to the server with the given credentials.
+ * If the connection is successful, the callback function is triggered.
+ * Otherwise, an error message is displayed.
+ * @param {String} serverURL URL of the server
+ * @param {String} credentials Credential to use for the connection
+ * @param {Function} callbackFunction The function to call if the connection is successful
+ */
 export const fetchCredentials = async (serverURL, credentials, callbackFunction) => {
-  fetch('https://' + serverURL + '/api/auth/login', {
-    method: 'POST',
+  try {
+    const response = await fetch(`https://${serverURL}/api/auth/login`, {
+      method: 'POST',
+      headers: {
+	Accept: 'application/json',
+	'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+	username: credentials.username,
+	password: credentials.password
+      })
+    })
+    if (response.status === 200) {
+      await callbackFunction(await response.json())
+    } else {
+      alert(i18n.t('Wrong credentials'))
+    }
+  } catch(error) {
+    alert(i18n.t('Something went wrong: ') + error.message)
+  }
+}
+
+/**
+ * Get the ID of the current user of a specific server using cache
+ * @param {String} serverURL URL of the server
+ * @returns The id of the connected user
+ */
+export const getCurrentUserID = async (serverURL) => {
+  let userId = -1
+
+  fetch(`https://${serverURL}/api/whoami`, {
```

Comment:
All the other functions use try...(await)...catch, I wonder if this one should do the same to be more coherent. Idon't have an strong opinion about it though, you can leave like that if you like it more.

Reply:
This one is now unused, I removed it.